parser/yaml/yaml.go

```diff
@@ -0,0 +1,831 @@
+//  Copyright 2013 Walter Schulze
```

Comment:
You can put your name and the year here :)

Reply:
Thanks!

Reply:
Thanks for the review. I've added unit tests that cover the most common cases based on the XML and JSON parsers. I'll definitely add a lot more as I make the YAML implementation more robust.